cocos/core/value-types/enum.ts

```diff
@@ -29,6 +29,24 @@ import { legacyCC } from '../global-exports';
 import { errorID } from '../platform/debug';
 import { assertIsTrue } from '../data/utils/asserts';
 
+type EnumType = Record<string, string | number>;
+
+interface EnumRegister {
+    /**
+     * `string` indicates the enumeration's name. `undefined` indicates the enumeration has no name.
+     */
+    name: string | undefined;
+
+    /**
+     * `null` if haven't not computed.
+     */
+    enumerators: null | Enum.Enumerator<EnumType>[];
+}
+
+const enumTypeRegistry = new WeakMap<EnumType, EnumRegister | null>();
```

Comment:
Why not just add a Symbol as an attribute directly? Like `enumTypeObj[Symbol("EnumRegister")] = registerObj`;

Reply:
It's not intrusive.